be/src/exec/es/es_scan_reader.cpp

```diff
@@ -127,7 +127,9 @@ Status ESScanReader::open() {
         RETURN_IF_ERROR(_network_client.init(_init_scroll_url));
         LOG(INFO) << "First scroll request URL: " << _init_scroll_url;
     }
-    _network_client.set_basic_auth(_user_name, _passwd);
+    if (!_user_name.empty() && !_passwd.empty()) {
```

Comment:
```suggestion
    if (!_user_name.empty() || !_passwd.empty()) {
```
If I'm not mistaken, basic authentication should only be set when both are empty.
So it seems to me that `||` should be used here.